### PR TITLE
AutoCompletion Feature for the CLI

### PR DESCRIPTION
**Pull request by AaronWB**
Divided the CLI up into four classes for simplicity of finding and modifying methods.

Added CLI Tests so we could test the CLI en-masse without having to launch the program each time, they check the data structure instead of the expected output.

Created an autocomplete feature that uses a predetermined list of commands to fill out and the user must press Tab (can be changed) to auto fill the word. The preview of the word being accepted by the user is on the right of the text box.

### CODE REVIEW

File: UML editor/pom.xml [hunk 1]
```diff
@@ -46,6 +46,12 @@
             <version>5.8.1</version>
             <scope>test</scope>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/jline/jline -->
+        <dependency>
+            <groupId>jline</groupId>
```
Comment (colbycrutcher) [inline UML editor/pom.xml:51]: jline dependency

File: UML editor/src/main/java/codecain/SaveManager.java [hunk 2]
```diff
@@ -37,18 +40,21 @@ public static void saveToJSON(String filePath) throws IOException {
...
      *
      * @param filePath The path of the JSON file to load.
+     * @throws IOException If an I/O error occurs while loading the file.
      */
     public static void loadFromJSON(String filePath) throws IOException {
         try {
             Map<String, Object> umlData = objectMapper.readValue(new File(filePath), Map.class);
 
-            UMLClass.classMap = (Map<String, UMLClassInfo>) umlData.get("classes");
-         //   UMLClass.classFields = (Map<Object, Map<Object, Object>>) umlData.get("fields");
```
Comment (colbycrutcher) [inline UML editor/src/main/java/codecain/SaveManager.java:50]: My changes, to fix some of the saves

File: UML editor/src/test/java/CLI_Tests.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        cli.getCommandManager().parseAndExecute("add class Address");
+        assertTrue(UMLClass.classMap.containsKey("Person") && UMLClass.classMap.containsKey("Address"), "Classes 'Person' and 'Address' should exist in the classMap.");
+    }
+
+    @Test
+    void testListRelationships() {
+        cli.getCommandManager().parseAndExecute("add class Person");
+        cli.getCommandManager().parseAndExecute("add class Address");
+        cli.getCommandManager().parseAndExecute("add relationship Person Address");
+        assertEquals(1, Relationship.relationshipList.size(), "There should be exactly one relationship in the relationshipList.");
```
Comment (colbycrutcher) [inline UML editor/src/test/java/CLI_Tests.java:150]: Some nice tests for edge cases

**Review by colbycrutcher — APPROVED** [review]
Looks good. The autocomplete is impressive. This adds a lot of ease of access, and works seamlessly. I think this will set us up well for sprint 3 requirements
